### PR TITLE
Add process for modifying the release notes

### PR DESCRIPTION
**Pull request by rogeruiz**
This was documented [in Slack][slack], so we should have it in our README for
folks on the team.

[slack]: https://hashicorp.slack.com/archives/CKE00RWUT/p1593026651157900

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+
+## Updating Release Notes (internal use only)
+
+Because of [the workflows in PTFE Releases][gh-ptfe-releases-ci]:lock: inability
+to overwrite files in this repository, updating release notes is a manual
+process. To update any release notes in this repository after they've been
+published, [submit a pull request][gh-new-pr] with the changes that need to be made.
+
+[gh-ptfe-releases-ci]: https://github.com/hashicorp/ptfe-releases/blob/master/.circleci/config.yml#L8-L21 "Config file for PTFE Releases"
+[gh-new-pr]: https://github.com/hashicorp/terraform-enterprise-release-notes/compare "Submit new PR in this repository"
```
Comment (erindatkinson) [inline README.md]: my only consideration is around the posting of the source repo in the public repo, but it's probably a moot point.

Comment (rogeruiz) [reply README.md]: hmm, so should I only reference it by name? that's a quick enough fix.

Comment (erindatkinson) [reply README.md]: yeah, I think that should be fine.

**Comment by rogeruiz** [comment]
Thanks @erindatkinson I've updated things and rebased the changes.

**Review by erindatkinson — APPROVED** [review]
LGTM 🌈

**Pull request merged**